src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -372,6 +376,8 @@ private AIState lookForNeededItems()
 
     /**
      * Updates the itemsCurrentlyNeeded with current values.
+     * 
+     * This function seems bugged, as it gives itemsCurrentlyNeeded = isEmpty 
```

Comment:
It is not empty just checked with a chatrequest of `items needed` in `syncNeededItemsWithInventory()`

Otherwhise I would be wondered. because if there were no needed items there would be nothing to build in the first place.

BUT i have found some strange behavior:
-> when removing the last line `itemsCurrentlyNeeded = new ArrayList<>(job.getItemsNeeded());` he will not build anything anymore -> not empty?
-> when iCN.isEmpty() right after its fill -> true so it is empty.

I checked usages of `itemsCurrentlyNeeded` with netbeans. ther is one place in the code which clears iCN but this is never called